packages/dropdown/src/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { css, cx } from '@leafygreen-ui/emotion';
+import {
+  useAutoScroll,
+  useBackdropClick,
+  useEventListener,
+  useIsomorphicLayoutEffect,
+  usePrevious,
+} from '@leafygreen-ui/hooks';
+import { RenderedContext } from '@leafygreen-ui/input-option';
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import Popover, { Align, Justify } from '@leafygreen-ui/popover';
+
+import { DescendantContext, useDescendants } from '../DescendantContext';
+import { DropdownContext } from '../DropdownContext';
+import { HighlightContext } from '../HighlightContext';
+import { handleKeyboardNavigation, useMergeRefs } from '../utils';
+
+import {
+  baseMenuStyle,
+  menuListStyle,
+  menuThemeStyles,
+} from './Dropdown.styles';
+import { DropdownProps, HighlightBehavior } from './Dropdown.types';
+
+export const Dropdown = React.forwardRef(
+  (
+    {
+      align = Align.Bottom,
+      adjustOnMutation = false,
+      children,
+      className,
+      darkMode: darkModeProp,
+      highlightBehavior = HighlightBehavior.Focus,
+      justify = Justify.End,
+      maxWidth,
+      open,
+      portalClassName,
+      portalContainer,
+      popoverZIndex,
+      scrollContainer,
+      setOpen,
+      shouldClose = () => true,
+      spacing = 6,
+      triggerRef,
+      usePortal = true,
+      renderedContext = RenderedContext.Menu,
+      ...rest
+    }: DropdownProps,
+    forwardRef,
+  ) => {
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+    const { ref, ...contextProps } = useDescendants();
+    const dropdownRef = useMergeRefs(forwardRef, ref);
+    const [highlightedRef, setHighlightedRef] = useState<HTMLElement | null>(
+      null,
+    );
+    const [firstOpen, setFirstOpen] = useState(false);
+    const previousOpenState = usePrevious(open);
+
+    // Gets list of registered item refs and filters out disabled items
+    // We use this to accurately cycle highlighted state
+    const enabledRefs = contextProps?.list?.current
+      ?.filter(element => {
+        const id = element._internalId;
+        return !contextProps?.map?.current?.[id].disabled;
+      })
+      .map(element => element.element);
+
+    // Keeps currently highlighted item in state
+    // And manually moves focus to appropriate item if `highlightBehavior` is set to 'focus'
+    const setHighlighted = useCallback(
+      (ref: HTMLElement | null) => {
+        setHighlightedRef(ref);
+
+        if (highlightBehavior === HighlightBehavior.Focus && ref) {
+          ref.focus();
+        }
+      },
+      [highlightBehavior],
+    );
+
+    // Check that this is the first time we're opening the Dropdown
+    // Will use this to move focus to the first element but the check below will be true before
+    // `enabledRefs` has been populated
+    useEffect(() => {
+      if (open && previousOpenState === false) {
```

Comment:
@TheSonOfThomp just added [lifecycle callbacks](https://github.com/mongodb/leafygreen-ui/pull/2067/files#diff-fa94af5fecd220d33eb35d775be5e80a1bda85636b20f7b7ad010f58df258183R97) to `Popover` so maybe you can leverage `onExited` to update `setFirstOpen` to `false` when the popover closes instead of using previous state and useEffect.